pkg/security/podsecuritypolicy/provider.go

```diff
@@ -296,6 +296,28 @@ func (s *simpleProvider) ValidatePod(pod *api.Pod) field.ErrorList {
 							"Flexvolume driver is not allowed to be used"))
 				}
 			}
+
+			if fsType == policy.CSI {
```

Comment:
I'm trying to remember why this was intended to behave differently than AllowedFlexVolumes (empty AllowedFlexVolumes means any flex volume is allowed).

Reply:
This was a safeguard against user requesting any CSI driver to function as inline ephemeral.  Based on suggestion above, it probably is best to make the CSI inline PSP work like other PSP's like FlexVolume (i.e. empty list means all driver allowed).  This will work since the existence of the `CSIDriver` object can further restrict that feature.

Reply:
Furthermore, if CSI PSP requires a non-empty allowed whitelist to work at all, then it creates a weird case where if `policy.CSI` is set and no volumes are provided, then it errors out.  For now, I think it's better to follow how FlexVol is used .

Reply:
> For now, I think it's better to follow how FlexVol is used .

sounds good. can you update the API doc